### PR TITLE
Add warning for smart placement w/assets

### PR DESCRIPTION
**Pull request by WillTaylorDev**
Fixes #0000

Provides a warning if using smart placement. Internal discussions still ongoing on whether to hard block on this instead of warn.

- Tests
  - [ ] TODO (before merge)
  - [X] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because: n/a
- Public documentation
  - [ ] TODO (before merge)
  - [X] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/18780/files
  - [ ] Documentation not necessary because:

### CODE REVIEW

**Comment by WillTaylorDev** [comment]
Smart placement docs here:
https://github.com/cloudflare/cloudflare-docs/pull/18780

File: packages/wrangler/src/assets.ts
Comment (WillTaylorDev) [inline packages/wrangler/src/assets.ts:453]: Update verbiage based on @korinne 's feedback, update with doc link

Comment (WillTaylorDev) [reply packages/wrangler/src/assets.ts:453]: `Warning: Turning on Smart Placement in a Worker that is using assets and `serve_directly` set to `false` means that your entire Worker could be moved to run closer to your data source, and all requests will go to that Worker before serving assets. This could result in poor performance as round trip times could increase when serving assets. Read more in the documentation here`

**Comment by WillTaylorDev** [comment]
- Needs changeset
- Needs to merge smartplacement docs first
- Need to wait until target branch is merged to main

File: packages/wrangler/src/assets.ts
Comment (rozenmd) [inline packages/wrangler/src/assets.ts:453]:
```suggestion
			"Turning on Smart Placement in a Worker that is using assets and serve_directly set to false means that your entire Worker could be moved to run closer to your data source, and all requests will go to that Worker before serving assets.\n" +
```

**Review by CarmenPopoviciu — APPROVED** [review]

**Review by penalosa — APPROVED** [review]

**Pull request merged**
